packages/rest/test/routes.test.ts

```diff
@@ -219,6 +219,32 @@ describe("rest: routes", () => {
     ]);
   });
 
+  it("models with non-required key property will produce required path parameter for key", async () => {
+    const [routes, diagnostics] = await compileOperations(`
+      using Cadl.Rest.Resource;
+
+      namespace Things {
+        model Thing {
+          @key
+          @segment("things")
+          thingId?: string;
```

Comment:
So should the key really be allowed to be optional?

Reply:
I'm not sure if there are cases where it's warranted, perhaps it's something that `@visibility` is better suited for.  @johanste would be a better person to ask about this.

Reply:
If this is getting too long to figure out if we need to allow optional @key, This PR shouldn't hurt and we can file na issue do discuss

Reply:
Sounds good, maybe I'll just go ahead and merge this to unblock the `cadl-azure` submodule update and then file an issue to discuss whether `@key` properties should always be required.

Reply:
Filed #736 and marked it as a design issue.